packages/react-components/react-datepicker-compat/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -385,6 +385,11 @@ export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HT
   root.onKeyDown = useEventCallback(mergeCallbacks(root.onKeyDown, onInputKeyDown));
   root.onFocus = useEventCallback(mergeCallbacks(root.onFocus, onInputFocus));
   root.onClick = useEventCallback(mergeCallbacks(root.onClick, onInputClick));
+  root.contentAfter = (
+    <span onClick={onIconClick} style={{ all: 'inherit' }}>
+      {root.contentAfter}
+    </span>
+  );
```

Comment:
Tested this locally and looks like this won't work as expected with styles plus adding a layer of DOM is not ideal. What need to be done here is take advantage that contentAfter is already a span. To achieve this we need to do the following:

(image of how the styles get a little messed up)
![image](https://github.com/microsoft/fluentui/assets/5953191/372e1ae7-c956-4913-9ea4-4365533cc632)

Resolve contentAfter ourselves to override the onClick prop. This isn't where we override it, but we need to resolve it this way so we can ensure the shorthand is not undefined and to add our default icon.
```jsx
    : 'outline';

  const [triggerWrapperRef, popupRef] = usePopupPositioning(props);

  // Resolving shorthand 👇
  const contentAfter = slot.always(props.contentAfter || {}, {
    defaultProps: {
      children: <CalendarMonthRegular />,
    },
    elementType: 'span',
  });

  const root = slot.always(restOfProps, {
    defaultProps: {
      appearance: inputAppearance,
```

next we have to merge the callback that was possibly provided with our icon onclick callback like this:

```jsx
const inputShorthand = slot.always(props.input, {
    elementType: 'input',
  });
  inputShorthand.ref = useMergedRefs(inputShorthand.ref, ref, rootRef);
  // merge callbacks 👇
  contentAfter.onClick = mergeCallbacks(onIconClick, contentAfter.onClick);
  // we replace the resolved shorthand with our contentAfter shorthand 👇
  root.contentAfter = contentAfter;

  root.input = inputShorthand;
  root.onChange = useEventCallback(mergeCallbacks(root.onChange, onInputChange));
```

sorry for having you make the previous change to then make this change 😥 this is a little more work, but it ensures reusability and makes the change in the DOM look cleaner :)

I've tested this locally, so you can go ahead and make these changes if you're still interested in contributing :) otherwise let me know and I can add the changes to your PR

Reply:
I like this a lot! And I have learned more about how the slots work :) Will make this change now

Reply:
Actually @sopranopillow I don't really care all that much about getting contributor credit. If you already have the changes, I'm totally ok with you pushing what you have :) Then I can see exactly what you have changed and use it as a learning exercise

Reply:
I think you'll still get the contribution :)

Reply:
Awesome, looks great :D thanks for the help!